### PR TITLE
5838 Additional Investment Size Filters

### PR DESCRIPTION
**Pull request by lshepstone**
@benlovell @esbel

### CODE REVIEW

File: app/serializers/firm_serializer.rb [hunk 1]
```diff
@@ -45,7 +47,15 @@ def other_advice_methods
     object.other_advice_method_ids
   end
 
+  def advises_on_investments
```
Comment (benlovell) [inline app/serializers/firm_serializer.rb]: No spec for this?

File: app/serializers/firm_serializer.rb [hunk 2]
```diff
@@ -45,7 +47,15 @@ def other_advice_methods
...
+    object.investment_size_ids.any?
```
Comment (tomas-stefano) [inline app/serializers/firm_serializer.rb]: This methods returns true or false. Can we change to use '?' on the end?

File: app/serializers/firm_serializer.rb [hunk 3]
```diff
@@ -45,7 +47,15 @@ def other_advice_methods
...
+  end
+
   def investment_sizes
     object.investment_size_ids
   end
+
+  def investment_transfers
+    object.pension_transfer_percent > 0
```
Comment (tomas-stefano) [inline app/serializers/firm_serializer.rb]: Same here.

File: app/serializers/firm_serializer.rb (hunk 2 above)
Comment (benlovell) [inline app/serializers/firm_serializer.rb]: It's in a serializer.

File: app/serializers/firm_serializer.rb (hunk 3 above)
Comment (tomas-stefano) [inline app/serializers/firm_serializer.rb]: Ow, it's a serializer. My bad. But I would move this to the object.

File: spec/serializers/firm_serializer_spec.rb [hunk 4]
```diff
@@ -40,6 +40,10 @@
       expect(subject[:investment_sizes]).to eql(firm.investment_size_ids)
     end
 
+    it 'exposes `investment_transfers`' do
+      expect(subject[:investment_transfers]).to be
```
Comment (tomas-stefano) [inline spec/serializers/firm_serializer_spec.rb]: Is there other context for this specs? If so, can we break into contexts?

File: app/serializers/firm_serializer.rb (hunk 2 above)
Comment (tomas-stefano) [inline app/serializers/firm_serializer.rb]: Yes, my bad :)

File: app/serializers/firm_serializer.rb (hunk 3 above)
Comment (benlovell) [inline app/serializers/firm_serializer.rb]: `object.pension_transfer_percent.nonzero?`

File: app/serializers/firm_serializer.rb (hunk 1 above)
Comment (lshepstone) [inline app/serializers/firm_serializer.rb]: Got rebased out like a knob - sorry about that.

**Comment by benlovell** [comment]
:+1:

**Comment by esbel** [comment]
:+1:

File: spec/serializers/firm_serializer_spec.rb [hunk 5]
```diff
@@ -40,6 +40,14 @@
...
     end
 
+    it 'exposes `advises_on_investments`' do
+      expect(subject[:advises_on_investments]).to be
+    end
+
+    it 'exposes `investment_transfers`' do
+      expect(subject[:investment_transfers]).to be
+    end
+
```
Comment (tomas-stefano) [inline spec/serializers/firm_serializer_spec.rb]: Will not have specs for when is false?

Comment (lshepstone) [inline spec/serializers/firm_serializer_spec.rb]: Short answer is that we build our queries to only ever match against `true` - but yeah, wouldn't hurt to have the additional coverage here.

Comment (benlovell) [inline spec/serializers/firm_serializer_spec.rb]: It's already covered. What would checking the inverse give us @tomas-stefano

Comment (benlovell) [inline spec/serializers/firm_serializer_spec.rb]: Unless you're talking about both cases, I would be more likely to agree about the first case. But even still, we'd just be checking that rails works correctly, IMO.

Comment (tomas-stefano) [inline spec/serializers/firm_serializer_spec.rb]: Is it already covered both cases? I tried to say e.g: if pension_transfer_percent is zero the API will returns false for that field. Or if investment_size_ids is empty will returns false. That both scenarios are covered?

**Pull request merged**